torch/distributed/_tensor/api.py

```diff
@@ -146,6 +146,10 @@ def forward(  # type: ignore[override]
             # TODO: by default check tensor metas across rank
             # TODO: See if we need to make this run_check logic
             # have a corresponding backward.
+
```

Comment:
you would need to do:
```
check_shape_stride = not shape and not stride
```
before passing this to `check_tensor_meta` instead of just blindly pass in `True`